src/app/util/ClientMonitoringRegistrationTable.h

```diff
@@ -0,0 +1,48 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceConfig.h>
+
+using MonitoringRegistrationStruct = chip::app::Clusters::ClientMonitoring::Structs::MonitoringRegistration::Type;
+
+namespace chip {
+class ClientMonitoringRegistrationTable
+{
+public:
+    /**********************************************************
+     * Functions Definitions
+     *********************************************************/
+
+    ClientMonitoringRegistrationTable() {}
+
+    static ClientMonitoringRegistrationTable & Instance();
+
+private:
+    /**********************************************************
+     * Attributes Declaration
+     *********************************************************/
```

Comment:
I would delete this and the comment above: uses a lot of vertical space without seeming to help me read the code better.

Reply:
done